crp_VIL/novatel_driver/novatel_gps_launcher/src/novatelTopicConverter.cpp

```diff
@@ -0,0 +1,28 @@
+#include "novatel_gps_launcher/novatelTopicConverter.hpp"
+
+
+crp::vil::NovatelTopicConverter::NovatelTopicConverter() : Node("novatel_topic_converter")
+{
+    m_sub_currentPose_ = this->create_subscription<geometry_msgs::msg::PoseStamped>("current_pose", 10, std::bind(&NovatelTopicConverter::currentPoseCallback, this, std::placeholders::_1));
+    
+    m_pub_currentPoseWithCovariance_ = this->create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>("/sensing/gnss/pose_with_covariance", 10);
+}
+
+
+void crp::vil::NovatelTopicConverter::currentPoseCallback(const geometry_msgs::msg::PoseStamped::SharedPtr msg)
+{
+    geometry_msgs::msg::PoseWithCovarianceStamped poseWithCovariance;
+    poseWithCovariance.header = msg->header;
+    poseWithCovariance.pose.pose = msg->pose;
+
+    m_pub_currentPoseWithCovariance_->publish(poseWithCovariance);
```

Comment:
I'm just wondering, if a) when current pose arrives, then publish pose  with covariance, or publish it in every e.g., 10 ms/20 ms, even if current pose is not received, and keep the last value till new current pose is received. What do you think @mesmatyi @AnonymDavid ?

Reply:
Defining a constant high publish rate would be nice so that all possible future GNSS devices would have the same output rate but it could also introduce additional difficulty to debugging because we cannot easily monitor the message rate after the driver. It's probably safer this way because any problems with the rate of the positioning data can be detected at every point of the dataflow. Although it probably won't be checked everywhere. Other concern is the introduced possible delay but at high enough publish rate it should be insignificant.

I don't see many advantages for this that would balance the mentioned possible problems. But I'm open for other opinions.